### DISCUSSION TITLE
Annotation syntax

### DISCUSSION THREAD

**Discussion by oleganza** [tact-lang/spec#9] [Ideas]
We might need a handy annotation syntax to wrap or attach metadata to types in compile time. 

Possible use cases:
* Adding interface impls,
* Adding serialization options,
* Adding actor type requirements (sender, gas, external/internal?)

Rust, Swift, Java have pretty much standard syntax for annotation (modulo punctuation):

```
@attr(foo, bar=baz)
blurp Flurp {
   ...
}
```

But since we have first-class compile-time execution I suggest we have annotations as a thin syntactic sugar for ordinary comptime function calls: similar to Ruby's shortcut block syntax: `foo(bar) { ... }` is equivalent to `foo(bar, proc{...})`.

Let's expand this discussion with concrete use-cases first, then follow up with the syntax bikeshed.

**Comment by p0lunin** [comment]
Use cases:
1. Compiler built-in modifiers. Such as `inline`, `external`/`internal`, `pure`/`impure`, `unsafe`, `layout`, `gas_limit`, etc.
2. Run-time pre/post calls when excecuting functions like `modifier` construction in solidity.
3. Possible can be used for (de)serialization.
4. Derives for the structs in Rust/Haskell way.
